### ISSUE TITLE
Plugin Details: Use the Dynamic catalog to get plugin data

### ISSUE THREAD

**Issue by WBerredo** [Automattic/wp-calypso#71323]
Use the Dynamic catalog on Jetpack Search to retrieve plugin data, instead of retrieving it from the .com and .org API as it's done currently.

It depends on: D95375-code

**Comment by cpapazoglou** [comment]
Duplicate of https://github.com/Automattic/dotcom-forge/issues/1482
